applications/poisson/sine/application.h

```diff
@@ -216,9 +217,26 @@ class Application : public ApplicationBase<dim, n_components, Number>
       }
     };
 
-    this->grid->create_triangulation(this->param.grid,
-                                     lambda_create_coarse_triangulation,
-                                     this->param.grid.n_refine_global);
+    auto const lambda_read_external_coarse_grid = [&](dealii::Triangulation<dim, dim> & tria) {
+      dealii::GridIn<dim> grid_in;
+
+      grid_in.attach_triangulation(tria);
+
+      grid_in.read_exodusii(this->input_parameters.input_grid_file);
+    };
+
+    if(this->input_parameters.read_external_grid)
+    {
+      this->grid->create_triangulation(this->param.grid,
```

Comment:
Wouldn't it make sense to introduce a version of `create_triangulation()` that accepts a file name?

Reply:
Yes, I could design an overloaded `create_triangulation()` function for reading an external grid, and get rid of the `lambda_read_external_coarse_grid` function in the application. I think it would be a better design. I will do this after the changes to the grid class are made.

Reply:
That would be great. Reading an external mesh is such a common operation! The lambda created here would just be created in the overloaded function!

Reply:
I agree. We do not want to rewrite these lines of code for every application reading a grid from a file. This functionality should instead become some generic functionality of ExaDG. In accordance with #320, we would realize this as a free function in `GridUtilities`.

What we have to think about in this context is how to deal with different file formats. We should avoid introducing again and again new functions with new names for different file endings. Instead, the function taking a filename as argument should internally deduce the type of grid, with an assert if the file-type is not implemented or incorrect.

Reply:
Why is there a need for a lambda in an overloaded function?

Reply:
I also didn't understand why a lambda would be needed.

Reply:
> What we have to think about in this context is how to deal with different file formats. We should avoid introducing again and again new functions with new names for different file endings. Instead, the function taking a filename as argument should internally deduce the type of grid, with an assert if the file-type is not implemented or incorrect.

To address this, we should use the [`grid_in.read(string,format)`](https://www.dealii.org/current/doxygen/deal.II/classGridIn.html#ad30d9d6f2ff9920c4285e795c4495826) function. But still, this function requires the file format as input. To solve this to options come two my mind: 

1. Adding a parameter to `input.json` and to `gridData` specifying the file format. 
2. Searching in the string for the file name extension.

Reply:
This reminds me of the different parameter file formats (json, prm, etc.). Some time ago, I introduced a function in deal.II which works for all the different input file formats, see https://github.com/dealii/dealii/pull/9741. I think we should strive for a similar solution in terms of reading different grid files (with different endings). This line shows how to extract the filename ending

https://github.com/dealii/dealii/blob/d77e5ebb0af4487fa3c503f97fd9d126f2eac18f/source/base/parameter_handler.cc#L573

So I think we should follow variant 2 and avoid an additional input parameter (variant 1), because this would again be redundant information.

Reply:
> What we have to think about in this context is how to deal with different file formats. We should avoid introducing again and again new functions with new names for different file endings. Instead, the function taking a filename as argument should internally deduce the type of grid, with an assert if the file-type is not implemented or incorrect.

This is done in 5b4d30d37476e26dfeab45c15911e31f36280fb3. Waiting for #320 to move the function to `GridUtilities`.

Reply:
Moved the function to `GridUtilities` in d932484523cc4b0a6ccbe41963e79a51e5d7419.